builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1231,6 +1325,12 @@ func (vm *virtualMachine) deployVirtualMachine(c *govmomi.Client) error {
 			return err
 		}
 	}
+
+	// Create the cdroms if needed.
+	if err := createCdroms(newVM, vm.cdroms); err != nil {
```

Comment:
One question. Shouldn't it call `createCdroms()` from both `createVirtualMachine()` and from `deployVirtualMachine()`?

Isn't `deployVirtualMachine()` used for cloning but `createVirtualMachine()` when creating a brand new vm?